mmpose/datasets/datasets/top_down/topdown_onehand10k_dataset.py

```diff
@@ -0,0 +1,244 @@
+            scale (np.ndarray[float32](2,)): scale of the bbox w & h.
+        """
+        aspect_ratio = self.ann_info['image_size'][0] / self.ann_info[
+            'image_size'][1]
+        center = np.array([x + w * 0.5, y + h * 0.5], dtype=np.float32)
+
+        if (not self.test_mode) and np.random.rand() < 0.3:
+            center += 0.4 * (np.random.rand(2) - 0.5) * [w, h]
+
+        if w > aspect_ratio * h:
+            h = w * 1.0 / aspect_ratio
+        elif w < aspect_ratio * h:
+            w = h * aspect_ratio
+
+        # pixel std is 200.0
+        scale = np.array([w / 200.0, h / 200.0], dtype=np.float32)
+
+        scale = scale * 1.25
+
+        return center, scale
+
+    def evaluate_kernal(self, pred, joints_3d, joints_3d_visible, bbox):
+        """Evaluate one example.
+
+        ||pre[i] - joints_3d[i]|| < 0.2 * max(w, h)
+        """
+        num_joints = self.ann_info['num_joints']
+        bbox = np.array(bbox)
+        threshold = np.max(bbox[2:]) * 0.2
+        hit = np.zeros(num_joints, dtype=np.float32)
+        exist = np.zeros(num_joints, dtype=np.float32)
+
+        for i in range(num_joints):
+            pred_pt = pred[i]
+            gt_pt = joints_3d[i]
+            vis = joints_3d_visible[i][0]
+            if vis:
+                exist[i] = 1
+            else:
+                continue
+            distance = np.linalg.norm(pred_pt[:2] - gt_pt[:2])
+            if distance < threshold:
+                hit[i] = 1
+        return hit, exist
+
+    def evaluate(self, outputs, res_folder, metrics='PCK', **kwargs):
+        """Evaluate OneHand10K keypoint results."""
+        res_file = os.path.join(res_folder, 'result_keypoints.json')
+
+        all_preds, all_boxes, all_image_path = list(map(list, zip(*outputs)))
+
+        kpts = []
+
+        for idx, kpt in enumerate(all_preds):
+            kpts.append({
+                'keypoints':
+                kpt[0],
+                'center':
+                all_boxes[idx][0][0:2],
+                'scale':
+                all_boxes[idx][0][2:4],
+                'area':
+                all_boxes[idx][0][4],
+                'score':
+                all_boxes[idx][0][5],
+                'image_id':
+                int(''.join(all_image_path[idx]).split('/')[-1][:-4]),
+            })
+
+        self._write_keypoint_results(kpts, res_file)
+        info_str = self._report_metric(res_file)
+        name_value = OrderedDict(info_str)
+
+        return name_value
+
+    def _write_keypoint_results(self, keypoints, res_file):
+        """Write results into a json file."""
+
+        with open(res_file, 'w') as f:
+            json.dump(keypoints, f, sort_keys=True, indent=4)
+
+    def _report_metric(self, res_file):
+        """Keypoint evaluation.
+
+        Report Mean Acc of skeleton, contour and all joints.
+        """
+        num_joints = self.ann_info['num_joints']
+        hit = np.zeros(num_joints)
```

Comment:
dtype

for all usage of np.zeros or ones, we prefer specify the dtype explicitly